### DISCUSSION TITLE
Cannot import ovpn file

### DISCUSSION THREAD

**Discussion by leviethung2103** [angristan/openvpn-install#1411] [Help]
![CleanShot 2024-11-11 at 09 01 17@2x](https://github.com/user-attachments/assets/4028aff7-c4d1-48df-a302-cdef80379e63)

After install the openvpn on Armbian Linux. It created the .ovpn file successfully. 

However, i cannot import the .ovpn file into the OpenVPN connect

Please suggest any solutions

**Comment by Mr0cket** [comment]
I believe this issue is because the command currently used to generate the TLS encryption key in the script `openvpn --genkey --secret /etc/openvpn/tls-crypt.key`, used here: https://github.com/angristan/openvpn-install/blob/e1f19e0f246c4e429dbf32bdf0a0a507d8ac194b/openvpn-install.sh#L781

is deprecated. shell logs below.

```
root@personal:~# openvpn --genkey --secret /etc/openvpn/tls-crypt.key
2024-11-25 00:02:03 DEPRECATED OPTION: The option --secret is deprecated.
2024-11-25 00:02:03 WARNING: Using --genkey --secret filename is DEPRECATED.  Use --genkey secret filename instead.
```

Fix is to update the offending lines in the script to use the new syntax for the command.

`openvpn --genkey secret /etc/openvpn/tls-crypt.key`

**Comment by Aveneid** [comment]
Probably your tls-key is not generated correctly. Is there empty ``` <tls-crypt> ``` in your ovpn file?

I fixed this by running script as root.
